src/NuGet.Core/NuGet.CommandLine.XPlat/Commands/PackageReferenceCommands/ListPackage/ListPackageCommandRunner.cs

```diff
@@ -414,15 +414,23 @@ private void PopulateSourceRepositoryCache(ListPackageArgs listPackageArgs)
             {
                 foreach (var topLevelPackage in frameworkPackages.TopLevelPackages)
                 {
-                    var matchingPackage = packageMetadata.Where(p => p.Key.Equals(topLevelPackage.Name, StringComparison.OrdinalIgnoreCase)).First();
+                    var matchingPackage = packageMetadata.Where(p => p.Key.Equals(topLevelPackage.Name, StringComparison.OrdinalIgnoreCase)).FirstOrDefault();
 
                     // Get latest metadata *only* if this is a report requiring "outdated" metadata
                     if (listPackageArgs.ReportType == ReportType.Outdated && matchingPackage.Value.Count > 0)
```

Comment:
There seems to be a bug here.  The `FirstOrDefault()` call will return the first matching element or `default(T)`.  Here, `T` is `System.Collections.Generic.KeyValuePair<string, List<string>>`, which is a struct.  If there is no matching element, `FirstOrDefault()` will return `default(KeyValuePair<string, List<string>>)` and the `Key` and `Value` properties for this object will both return `null`.  That means the subsequent call to `matchingPackage.Value.Count` would throw `NullReferenceExcepton`.

To see what I mean, try running this code on https://try.dot.net:

```C#
using System;
using System.Collections.Generic;
using System.Linq;

public class Program
{
  public static void Main()
  {
    Dictionary<string, List<string>> dictionary = new Dictionary<string, List<string>>();
    KeyValuePair<string, List<string>> keyValuePair = dictionary.FirstOrDefault();

    Console.WriteLine($"Key == {keyValuePair.Key ?? "(null)"}");
    Console.WriteLine($"Value == {(keyValuePair.Value is null ? "(null)" : keyValuePair.Value.Count.ToString())}");
  }
}
```

Reply:
I've not been able to test this, do you have any suggestions on how can I test it? 

Since this function is not public and static, I cannot create a unit test and it's been hard to create a more complex test with mocked data.

Reply:
> I've not been able to test this, do you have any suggestions on how can I test it?

The goal I understand is to write a test such that `matchingPackage` becomes `null`. Normally, that'd be as simple as creating a mocked packageMetadata which will have 0 results, and therefore your code will return `null`. However, I'm not sure with all these test "helper" contexts exactly how that works. Maybe you already know how metadata is being pulled from the test, but if not, I'd start there.

Reply:
Added the test!